diff --git a/src/com/junhzhan/cal/widget/CalendarWidgetNew.java b/src/com/junhzhan/cal/widget/CalendarWidgetNew.java
index a20ee20..750ae97 100644
--- a/src/com/junhzhan/cal/widget/CalendarWidgetNew.java
+++ b/src/com/junhzhan/cal/widget/CalendarWidgetNew.java
@@ -83,6 +83,8 @@ public class CalendarWidgetNew extends LinearLayout {
     
     private HashMap<CalendarItem, Set<Integer>> mHasExtraItems = new HashMap<CalendarItem, Set<Integer>>();
     
+    private boolean mViewPagerScrolling = false;
+    
 
     public CalendarWidgetNew(Context context, AttributeSet attrs, int defStyle) {
         super(context, attrs, defStyle);
@@ -301,6 +303,20 @@ public class CalendarWidgetNew extends LinearLayout {
             super(context);
             setOrientation(VERTICAL);
         }
+        
+        
+        
+
+        @Override
+        public boolean dispatchTouchEvent(MotionEvent ev) {
+            if (ev.getAction() == MotionEvent.ACTION_DOWN && mViewPagerScrolling) {
+                return false;
+            }
+            return super.dispatchTouchEvent(ev);
+        }
+
+
+
 
         @Override
         public boolean onInterceptTouchEvent(MotionEvent ev) {
@@ -406,6 +422,7 @@ public class CalendarWidgetNew extends LinearLayout {
             } else {
                 mHandler.removeCallbacks(mRunnable);
             }
+            Log.e(TAG, "target row index is " + mTargetRow);
             if (deltaY > 0) {
                 if (getScrollY() > 0) {
                     if ((getScrollY() - deltaY) <= 0) {
@@ -420,7 +437,7 @@ public class CalendarWidgetNew extends LinearLayout {
                         }
                     }
                 } else {
-                    if (mResizableContainer.getHeight() < CalendarViewEfficient.ROW_COUNT * mRowHeight) {
+                    if (mResizableContainer.getHeight() <= CalendarViewEfficient.ROW_COUNT * mRowHeight) {
                         if ((mResizableContainer.getHeight() + deltaY) >= CalendarViewEfficient.ROW_COUNT * mRowHeight) {
                             mResizableContainer.getLayoutParams().height = CalendarViewEfficient.ROW_COUNT * mRowHeight;
                             mResizableContainer.requestLayout();
@@ -1472,6 +1489,7 @@ public class CalendarWidgetNew extends LinearLayout {
         public void onPageScrollStateChanged(int state) {
             mViewPagerScrollState = state;
             if (state == ViewPager.SCROLL_STATE_IDLE) {
+                mViewPagerScrolling = false;
                 int current = mCalendarPager.getCurrentItem();
                 if (current == 0) {
                     mHandler.post(new Runnable() {
@@ -1493,6 +1511,8 @@ public class CalendarWidgetNew extends LinearLayout {
                         }
                     });
                 }
+            } else {
+                mViewPagerScrolling = true;
             }
         }
     };
